From e69f18283c8f6722b8a3eff2d4602df08d3e3330 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Mon, 27 Dec 2021 07:38:46 +0200
Subject: [PATCH] [SW-68327] gaudi2: clear cpld last reset cause

After prstn, watchdog or ecc serr and derr cpld firmware informs
last reset cause using GPIO_66 and GPIO_67. Once ppboot has read
these gpios it needs to clear up this cause in cpld. This is done
using GPIO_68. Currently clearing this reset source is not working
in CPLD_REV 0x94. With CPLD firmware revision 0xA4 which has fixes
for GPIO_68, this patch writes 0x0 and then 0x1 to GPIO_68 to clear
last reset cause.

Change-Id: Ic266a911a8d273a11e0664354dc16538ee205c85
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 2dff75a..d863c59 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1049,6 +1049,7 @@
 	cpld_rst_src = (rst_src_1 << 1) | rst_src_0;
 
 	/* clear CPLD reset src event */
+	gaudi2_gpio_write(GAUDI2_RST_SRC_CLR_GPIO, 0x0);
 	gaudi2_gpio_write(GAUDI2_RST_SRC_CLR_GPIO, 0x1);
 
 	return cpld_rst_src;
